include/revng/Pipeline/InvalidationEvent.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+namespace Pipeline {
+
+class InvalidationEventBase {
+private:
+  const char *ID;
+
+public:
+  InvalidationEventBase(char &ID) : ID(&ID) {}
+  virtual ~InvalidationEventBase() = default;
+
+  template<typename Derived>
+  bool isA() const {
+    return ID == &Derived::ID;
+  }
+};
+
+template<typename Derived>
+class InvalidationEvent : public InvalidationEventBase {
+public:
+  InvalidationEvent() : InvalidationEventBase(Derived::ID) {}
+
+  static bool classof(const InvalidationEventBase *Instance) {
+    return Instance->isA<Derived>();
+  }
```

Comment:
yeah i only erased the destruct but it's because we have not decided what to do with invalidation events yet. i guess i can drop them entirely and add them in the future

Reply:
If you think it's not much work, OK.
Maybe throw the files somewhere for future reference.